scipy/stats/_stats_py.py

```diff
@@ -440,6 +440,12 @@ def mode(a, axis=0, nan_policy='propagate'):
     The mode of object arrays is calculated using `collections.Counter`, which
     treats NaNs with different binary representations as distinct.
 
+    .. deprecated:: 1.9.0
+        Support for object arrays has been deprecated and will be removed in
+        the second release after SciPy 1.9.0.
+        `pandas.DataFrame.mode <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.mode.html>`_
+        can be used instead.
+
     The mode of arrays with other dtypes is calculated using `np.unique`.
```

Comment:
To be clear, this is what I am proposing/wondering about. The link in the warning in the code is fine

```suggestion
    treats NaNs with different binary representations as distinct.

    The mode of arrays with other dtypes is calculated using `np.unique`.
```

Reply:
I was just following gh-15512. I find a short deprecation note in docs useful. I think these could be helpful as a reminder to remove the support later (some deprecation warnings easily get lost in the code).

Reply:
Sure I get and agree with the rational, as I said I am just -0.5 but fine if it goes in. I will let Matt be the judge 😉

Reply:
Meh I don't see a problem with it. I think it's helpful and renders fine.
![image](https://user-images.githubusercontent.com/6570539/153725426-028b0518-52ff-438d-babf-6f59a7f74b62.png)